### ISSUE TITLE
FAILED! => {"changed": false, "msg": "AnsibleUndefinedVariable: 'ansible.vars.hostvars.HostVarsVars object' has no attribute 'zookeeper_id'"}

### ISSUE THREAD

**Issue by Zaimeth** [sleighzy/ansible-zookeeper#20]
Hello, i'm currently use your ansible-galaxy playbook in my local vm, and that error showing, message error : 

```fatal: [target1]: FAILED! => {"changed": false, "msg": "AnsibleUndefinedVariable: 'ansible.vars.hostvars.HostVarsVars object' has no attribute 'zookeeper_id'"}```

And after i check it, i thinks because u change code in this PR : https://github.com/sleighzy/ansible-zookeeper/pull/13

and, when i change the code locally from :

```server.{{ hostvars[host].zookeeper_id }}={{ hostvars[host].ansible_nodename }}:{{ zookeeper_leader_port }}:{{ zookeeper_election_port }}```

to

```server.{{ hostvars[host].zookeeper_id | default(zookeeper_id) }}={{ hostvars[host].ansible_nodename }}:{{ zookeeper_leader_port }}:{{ zookeeper_election_port }}```

the error not showing again, and my playbook success. does that mean you have to rollback the code of this section?

**Comment by sleighzy** [comment]
Hi @Zaimeth . Thanks for pointing that out. Looks to be a simple change to just add the usage of the `default(zookeeper_id)` back in again if that variable isn't valued.

I can fix this up in the next couple of days unless you have something for a PR already. Let me know.

In the interim you can just specify the `zookeeper_id` variable in your playbook, host/group vars, etc. so that it's populated if you want to run this without modifying the role.

**Comment by Zaimeth** [comment]
Okay, thanks @sleighzy . I will wait for this to be fixed, and in the meantime I will hardcode it locally.
I have tried using variables as suggested by you, but the results are the same, the error output still appears

**Comment by sleighzy** [comment]
Hi @Zaimeth . I have fixed this now and released version 1.2.1 of this to Ansible Galaxy as well. Thanks for the report.

**Comment by sleighzy** [comment]
> I have tried using variables as suggested by you, but the results are the same, the error output still appears

My apologies. Would just be set in host_vars files or could be specified in your inventory (easier). For example in your inventory file.

```
[zookeeper-nodes]
zookeeper-1 zookeeper_id=1
zookeeper-2 zookeeper_id=2
zookeeper-3 zookeeper_id=3
```

Even with the fix you will still need to be using something like this if you have more than one node in your cluster.
